Dockerfile

```diff
@@ -1,37 +1,37 @@
-# Settings.
+# Build arguments.
 ARG USER_ID=3001
 ARG USER_NAME=vireo
-ARG SOURCE_DIR=/$USER_NAME/source
+ARG HOME_DIR=/$USER_NAME
+ARG SOURCE_DIR=$HOME_DIR/source
+ARG APP_PATH=/var/vireo
 ARG NODE_ENV=production
 
 # Maven stage.
-FROM maven:3-openjdk-11-slim as maven
+FROM maven:3-eclipse-temurin-11-alpine as maven
 ARG USER_ID
 ARG USER_NAME
+ARG HOME_DIR
 ARG SOURCE_DIR
+ARG APP_PATH
 ARG NODE_ENV
 
 ENV NODE_ENV=$NODE_ENV
 
-# Create the user and group (use a high ID to attempt to avoid conflicts).
-RUN groupadd --non-unique -g $USER_ID $USER_NAME && \
-    useradd --non-unique -d /$USER_NAME -m -u $USER_ID -g $USER_ID $USER_NAME
+# Create the group (use a high ID to attempt to avoid conflits).
+RUN addgroup -g $USER_ID $USER_NAME
```

Comment:
What is the reason for removing `--non-unique`?
This was done in other repositories to ensure this does not fail.

Also, this `addgroup` and that `adduser` line below were combined for performance reasons. Are we throwing this performance reason away now? Why? Is the a bug in regards to this?

Reply:
I believe `--non-unique` is added here:
- https://github.com/TAMULib/CAP/pull/186

I believe the reason was because of possible existing conflicts in the docker system.

For that `groupadd` and `useradd` grouping them together is what you had advocated for, for performance reasons. I simple have not objected.